openqa-label-known-issues

```diff
@@ -134,14 +134,16 @@ investigate_issue() {
     # against even in case when autoinst-log.txt is missing the details, e.g.
     # see https://progress.opensuse.org/issues/69178
     echo "$reason" >> "$out"
-    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]]; then
+    # Control flow when there is a curl issue but still want to process job due to reason var
+    # even if autoinst-log.txt not found but assume is missing.
+    [[ -n $reason ]] && [[ $curl_output = "404" ]] && [[ $reason != null ]] \
+        && is_unreachachle_but_has_reason=1 \
+        || is_unreachachle_but_has_reason=0
+    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]] && [[ ! $is_unreachachle_but_has_reason ]]; then
         # if we can not even access the page it is something more critical
         handle_unreachable "$testurl" "$out"
-        # Assume autoinst-log.txt still not found but process job anyway due to reason var
-        if [[ -n $reason ]] && [[ $curl_output = "404" ]] && [[ $reason != null ]]; then
-            label_on_issues_from_issue_tracker "$id"
         # Checking timestamp
-        elif [[ $(date -uIs -d '-14days') > $(grep timeago "$out" | hxselect -s '\n' -c '.timeago::attr(title)') ]]; then
+        if [[ $(date -uIs -d '-14days') > $(grep timeago "$out" | hxselect -s '\n' -c '.timeago::attr(title)') ]]; then
```

Comment:
I am not sure if I read all the conditions correctly, but within this block you need non-standard curl 
 output and `$is_unreachachle_but_has_reason == false`, which effectively means `$reason == ""` or `$reason == null`. `$out` is basically curl output plus `$reason`, which is in this special case empty or almost empty (as there is no `$reason`), that means `grep` can't do much in this case. Maybe this block needs to be one level higher?

Reply:
I am not sure about that. This one is not modified at all from my side. and it is not directly correlated to the problem this commit tries to fix I think.

Reply:
Well, you did change the condition here 😉

Reply:
IIUC you are right. `$out` =  curl output plus `$reason` if any. However the condition is activated only for old jobs which found older than 14d. And only checked when the curl is not successful. Not successful means that it couldnt get the `autoinst-log.txt`.
Currently if the `autoinst-log.txt` not found but `$is_unreachachle_but_has_reason` is TRUE, the `label_on_issues_from_issue_tracker` branch should run

Reply:
I guess it is OK then? Would be nice to have it covered by some automated tests.

Reply:
I consider this closed as we have added tests. Lets me know in the latest commit.